src/app/_hooks/useDebounce.js

```diff
@@ -11,3 +11,21 @@ export function useDebounce(state, ms) {
   }, [state, ms]);
   return debouncedState;
 }
+
+export function useDebounceCallback(fn, timeout) {
+  const timeoutRef = useRef(null);
+  return useCallback(
+    (...args) => {
+      // Clear the current timeout, if any, to reset the debouncing timer.
+      if (timeoutRef.current !== null) {
+        clearTimeout(timeoutRef.current);
+      }
+
+      // Set a new timeout to invoke the original function after the delay.
+      timeoutRef.current = setTimeout(() => {
```

Comment:
Try to use as less comments as possible. Especially in cases when you describe what you do, not why. Code is always self-descriptive

Reply:
thanks, will do